apps/nova/src/components/Settings/ProfileForm.tsx

```diff
@@ -0,0 +1,352 @@
+import { Controller, Form, Label } from "@litespace/luna/Form";
+import UploadPhoto from "@/components/Settings/UploadPhoto";
+import { Typography } from "@litespace/luna/Typography";
+import { FullSwitch } from "@litespace/luna/Switch";
+import { Button, ButtonSize } from "@litespace/luna/Button";
+import { useFormatMessage } from "@litespace/luna/hooks/intl";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { useForm } from "react-hook-form";
+import { IUser } from "@litespace/types";
+import { useToast } from "@litespace/luna/Toast";
+import { orNull, orUndefined } from "@litespace/sol/utils";
+import { useInvalidateQuery } from "@litespace/headless/query";
+import {
+  useRequired,
+  useValidateEmail,
+  useValidatePassword,
+  useValidatePhoneNumber,
+  useValidateUsername,
+} from "@litespace/luna/hooks/validation";
+import { QueryKey } from "@litespace/headless/constants";
+import { useUpdateUser } from "@litespace/headless/user";
+import TopicSelector from "@/components/Settings/TopicSelector";
+import { isEmpty, isEqual } from "lodash";
+
+type IForm = {
+  name: string;
+  email: string;
+  phoneNumber: string;
+  password: {
+    new: string;
+    current: string;
+    confirm: string;
+  };
+  city: IUser.City;
+};
+
+export type TopicWatcher = {
+  addTopics: number[];
+  removeTopics: number[];
+};
+
+export const ProfileForm: React.FC<{ user: IUser.Self }> = ({ user }) => {
+  const [photo, setPhoto] = useState<File | null>(null);
+  const [topicChanges, setTopicChanges] = useState<TopicWatcher>({
+    addTopics: [],
+    removeTopics: [],
+  });
```

Comment:
- No need to maintain a state of the added & removed topics. You can have a state for all topics that the user selected and when you are about to submit the data you compare the initial topics you got form the backend with the topics you have after that performed his changes. 
- The topics can be part of the form state. 
- Note that max topics that the user can select is 30. You can find this constant [here](https://github.com/litespace-org/litespace/blob/a65cd9a2b3859ed3aabd2b9f4ee62903d4d2ef59/packages/sol/src/constants.ts#L65)